src/Shared/E2ETesting/WaitAssert.cs

```diff
@@ -57,16 +61,24 @@ private static void WaitAssertCore(IWebDriver driver, Action assertion, TimeSpan
                         assertion();
                         return true;
                     }
-                    catch
+                    catch(Exception e)
                     {
+                        lastException = e;
                         return false;
                     }
                 });
             }
             catch (WebDriverTimeoutException)
             {
-                // Instead of reporting it as a timeout, report the Xunit exception
-                assertion();
+                if (lastException != null)
+                {
+                    ExceptionDispatchInfo.Capture(lastException).Throw();
+                }
+                else
+                {
+                    // Instead of reporting it as a timeout, report the Xunit exception
+                    assertion();
+                }
```

Comment:
Can you clarify why it's advantageous to do this? I recall we discussed this before. What reason is there to prefer `lastException` over invoking `assertion` again? This logic seems really arbitrary and misleading about its purpose.

Reply:
Rerunning the assertion might make it fail for other reasons.

Reply:
We're repeating the same discussion from last time :)

If we run the assertion 15 times and it fails each time, then we run it for a 16th time and it fails again, I don't see why we should regard the 15th exception as any more "correct" than the 16th or the 1st or any other in the sequence.

Reply:
I believe 15th is equally incorrect to 1st, but more correct than 16th. If you feel strongly about it, I'll revert it (i'm doing some changes in spa templates, so I can do it while I'm there).

Reply:
Nah, on further consideration, it doesn't really matter. There'll be no observable difference, so either way is fine. Sorry for raising it.